packages/hierarchies-react/src/presentation-hierarchies-react/TreeNodeRenderer.tsx

```diff
@@ -107,21 +123,69 @@ function PlaceholderNode(props: Omit<TreeNodeProps, "onExpanded">) {
   return <TreeNode {...props} icon={<ProgressRadial size="x-small" indeterminate />} onExpanded={/* istanbul ignore next */ () => {}}></TreeNode>;
 }
 
-function ResultSetTooLargeNode({ onRemoveLimit, ...props }: Omit<TreeNodeProps, "onExpanded"> & { onRemoveLimit: () => void }) {
+function ResultSetTooLargeNode({
+  onFilterClick,
+  onOverrideLimit,
+  limit,
+  ...props
+}: Omit<TreeNodeProps, "onExpanded" | "label"> & ResultSetTooLargeNodeLabelProps) {
   return (
-    <TreeNode {...props} onExpanded={/* istanbul ignore next */ () => {}}>
-      <Button
-        styleType="borderless"
-        size="small"
-        onClick={(e) => {
-          onRemoveLimit();
-          e.stopPropagation();
-        }}
-      >
-        Remove Limit
-      </Button>
-    </TreeNode>
+    <TreeNode
+      {...props}
+      className="stateless-tree-node"
+      label={<ResultSetTooLargeNodeLabel limit={limit} onFilterClick={onFilterClick} onOverrideLimit={onOverrideLimit} />}
+      onExpanded={/* istanbul ignore next */ () => {}}
+    />
   );
 }
 
-type TreeNodeProps = ComponentPropsWithoutRef<typeof TreeNode>;
+interface ResultSetTooLargeNodeLabelProps {
+  limit: number;
+  onFilterClick: () => void;
+  onOverrideLimit: (limit: number) => void;
+}
+
+function ResultSetTooLargeNodeLabel({ onFilterClick, onOverrideLimit, limit }: ResultSetTooLargeNodeLabelProps) {
+  return (
+    <Flex
+      flexDirection="column"
+      gap="3xs"
+      title={`Please provide additional filtering - there are more items than allowed limit of ${limit}. Increase hierarchy level limit`}
+      alignItems="start"
+    >
+      <Flex flexDirection="row" gap="3xs">
+        <Text>Please provide</Text>
+        <Anchor
+          underline
+          onClick={(e) => {
+            e.stopPropagation();
+            onFilterClick();
+          }}
+        >
+          additional filtering
+        </Anchor>
+        <Text>- there are more items than allowed limit of {`${limit}`}.</Text>
+      </Flex>
+      {limit < MAX_LIMIT_OVERRIDE ? (
+        <Flex flexDirection="row" gap="3xs">
+          <Anchor
+            underline
+            onClick={(e) => {
+              e.stopPropagation();
+              onOverrideLimit(getLimitOverride(limit));
+            }}
+          >
+            Increase hierarchy level limit
+          </Anchor>
+        </Flex>
+      ) : null}
+    </Flex>
+  );
+}
+
+function getLimitOverride(current: number) {
+  if (current < MAX_LIMIT_OVERRIDE / 2) {
+    return MAX_LIMIT_OVERRIDE / 2;
+  }
+  return MAX_LIMIT_OVERRIDE;
+}
```

Comment:
I think for simplicity we should only allow overriding the limit once. We should also mention what the override is in the `Anchor`.